### PR TITLE
Fixed things based on list of found bugs

### LINKED ISSUE #89: Small fixes from testing round 1

### PR DESCRIPTION
**Pull request by lauriLukkarinen**
Resolves #89

Addresses at least the following:

### Roles-page

Wrong header: Users, should be Roles
> changed

Both roles and positions in use as a term
> only position is used

Icon-cell should not be editable, if the icon cannot be edited
> removed all editable fields as we did not have the functionality in the first place

### Filters drawer

If user closes filters drawer after clicking Clear but without clicking Apply, the filters stay but don’t show up anymore in the filters-drawer
> drawer now closes after clicking empty

### Milestones

Choosing current day as the starting day makes the duration a very long decimal number
> rouded to full days

### Creating tasks

Page has to be reloaded between creating or editing tasks
> fixed

### Change proposals

Accepting a change proposal should update the list immediately, not only after reloading the page
> fixed

Change proposals don’t show up in the task dialog when on Tasks-page
> fixed

### CODE REVIEW

File: src/components/users/new-user-dialog.tsx [hunk 1]
```diff
@@ -47,14 +47,15 @@ const NewUserDialog = ({ projectId }: Props) => {
   const [selectedProject, setSelectedProject] = useState<Project>();
   const [selectedJobPosition, setSelectedJobPosition] = useState<JobPosition>();
 
-  const listUsersQuery = useListUsersQuery();
+  const listUsersQuery = useListUsersQuery(undefined, { enabled: open });
```
Comment (Jdallos) [inline src/components/users/new-user-dialog.tsx:50]: Could this break any other list all user requests?

File: src/routes/projects.tsx [hunk 2]
```diff
@@ -85,7 +87,7 @@ function ProjectsIndexRoute() {
         </Box>
       </Toolbar>
       <Card sx={{ flex: 1, minWidth: 0 }}>
-        <DataGrid
+        <DataGrid<Project>
```
Comment (Jdallos) [inline src/routes/projects.tsx:90]: This looks weird?

File: src/theme/index.ts [hunk 3]
```diff
@@ -1,4 +1,5 @@
 import { createTheme } from "@mui/material";
+import type {} from "@mui/x-data-grid/themeAugmentation";
```
Comment (Jdallos) [inline src/theme/index.ts:2]: Is this needed?

**Review by Jdallos — APPROVED** [review]

File: src/routes/projects.tsx (hunk 2 above)
Comment (lauriLukkarinen) [reply src/routes/projects.tsx:90]: This is to provide specific type for the component. It automatically infers the column definition typings from that.

File: src/theme/index.ts (hunk 3 above)
Comment (lauriLukkarinen) [reply src/theme/index.ts:2]: It is, as this provides MuiDataGrid theme types via type augmentation to ThemeOptions, which is now required in createTheme where we provide the default theming to the DataGrid.

**Review by Jdallos — APPROVED** [review]

**Pull request merged**